### PR TITLE
RFR: made several queries to display data in defferent sorts on homepage and ...

### PR DESCRIPTION
**Pull request by RBIII**
...SitcomsManager to handle most of the logic

### CODE REVIEW

File: app/controllers/sitcoms_controller.rb [hunk 1]
```diff
@@ -4,9 +4,22 @@ class SitcomsController < ApplicationController
...
+      @most_reviewed = sitcom_manager_with_search.most_reviewed
     else
+      sitcom_manager_without_search = SitcomManager.new(nil, params[:page])
+
       @sitcoms = Sitcom.order(start_year: :desc).page params[:page]
+
+      @newest_sitcoms = sitcom_manager_without_search.newest_sitcoms
+      @highest_rated = sitcom_manager_without_search.highest_rated
+      @most_reviewed = sitcom_manager_without_search.most_reviewed
+      binding.pry
```
Comment (SpencerCDixon) [inline app/controllers/sitcoms_controller.rb]: remove binding.pry

File: app/models/sitcom_manager.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
+class SitcomManager
+  attr_reader :search_params, :page_params
+
+  def initialize(search_params, page)
+    @search_params = search_params
+    @page_params = page_params
```
Comment (SpencerCDixon) [inline app/models/sitcom_manager.rb:6]: mispelled page_params with page here I think?

File: app/controllers/sitcoms_controller.rb [hunk 3]
```diff
@@ -4,9 +4,21 @@ class SitcomsController < ApplicationController
 
   def index
     unless params[:search] == nil
```
Comment (dtnorris) [inline app/controllers/sitcoms_controller.rb:7]: This controller action is looking super fat with a ton of instance variables. Could you refactor to clean that up? If you need ideas I'm happy to provide some.

File: app/models/sitcom_manager.rb [hunk 4]
```diff
@@ -0,0 +1,71 @@
+class SitcomManager
```
Comment (dtnorris) [inline app/models/sitcom_manager.rb:1]: can you add unit tests for this class?

**Pull request merged**